src/widget/wtracktableview.cpp

```diff
@@ -1597,6 +1603,22 @@ void WTrackTableView::slotClearBeats() {
     }
 }
 
+void WTrackTableView::slotClearWaveform() {
+    TrackModel* trackModel = getTrackModel();
+    if (trackModel == NULL) {
+        return;
+    }
+
+    QModelIndexList selectedIndices = selectionModel()->selectedRows();
+    foreach (QModelIndex index, selectedIndices) {
+        TrackPointer pTrack = trackModel->getTrack(index);
+        if (pTrack) {
+            pTrack->setWaveform(WaveformPointer());
```

Comment:
I've done some thinking (that's why I'm not answering for two days :smile:) and I like your idea for adding a flag to `Waveform`. IMHO it is much cleaner than having a NULL pointer. But how do I name it? `m_bNeedsToBeCleared` seems a little too long and `m_bInvalid` too ambiguous (`Waveform` already has an `isValid` method).